package.py

```diff
@@ -0,0 +1,911 @@
+        self.__keychain = 'rotki-build.keychain'
+        self.__p12 = '/tmp/certificate.p12'
+
+    @staticmethod
+    def unpack_wheels(
+            package_version: str,
+            plt: Literal['macosx_10_9_x86_64', 'macosx_11_0_arm64'],
+            directory: Path,
+    ) -> None:
+        directory.mkdir(exist_ok=True)
+        os.chdir(directory)
+        subprocess.call(
+            f'pip download {package_version} --platform {plt} --only-binary=:all:',
+            shell=True,
+        )
+        for file in directory.iterdir():
+            if file.name.find(plt) >= 0:
+                logger.info(f'unpacking wheel {file}')
+                subprocess.call(f'wheel unpack {file}', shell=True)
+
+            os.unlink(file)
+
+    @staticmethod
+    def macos_link_archs(source: Path, destination: Path) -> None:
+        logger.info(f'creating fat binary {source} <-> {destination}')
+        ret_code = subprocess.call(
+            f'lipo -create -output {destination} {source} {destination}',
+            shell=True,
+        )
+
+        if ret_code != 0:
+            logger.error(f'failed to create a fat binary {source} {destination}')
+            exit(1)
+
+        archs = subprocess.check_output(f'lipo -archs {destination}', encoding="utf-8", shell=True)
+
+        if archs.strip() != 'x86_64 arm64':
+            logger.error(f'{destination} was not a fat binary, only has {archs}')
+            exit(1)
+
+    @staticmethod
+    def modify_wheel_metadata(wheel_metadata: Path) -> None:
+        with open(wheel_metadata, 'r') as file:
+            data = file.readlines()
+            for (index, line) in enumerate(data):
+                if not line.startswith('Tag'):
+                    continue
+                data[index] = line.replace('x86_64', 'universal2')
+
+        with open(wheel_metadata, 'w') as file:
+            file.writelines(data)
+
+    def __download_patched_pip(self) -> Path:
+        pip_wheel = 'pip-22.1.2-py3-none-any.whl'
+        response = requests.get(f'https://github.com/rotki/rotki-build/raw/main/{pip_wheel}')
+        if response.status_code == 200:
+            temporary_directory = self.__storage.temporary_directory
+            temporary_directory.mkdir(exist_ok=True)
+            wheel_file = temporary_directory.joinpath(pip_wheel)
+            with open(wheel_file, 'wb') as file:
+                file.write(response.content)
+            return wheel_file
+
+        logger.error(f'pip failed to download {pip_wheel}')
+        raise Exception(f'{pip_wheel} download failed')
+
+    def __get_versions(self, packages: list[str]) -> dict[str, str]:
+        requirements = self.__storage.working_directory.joinpath('requirements.txt')
+        package_versions: dict[str, str] = {}
+        with open(requirements) as fp:
+            while True:
+                line = fp.readline()
+
+                if not line:
+                    break
+
+                if not line.strip() or line.startswith('#'):
```

Comment:
Better be doing explicit checks. Here I believe you want to test if the line is not empty so 
```suggestion
                if len(line.strip()) != 0 or line.startswith('#'):
```

Reply:
This skips empty lines or comments in the requirements, so it would be `if len(line.strip()) == 0`

Reply:
yes sorry, was thinking and mixed it